src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -147,39 +149,59 @@ def prepare_or_run_pipeline(
         Raises:
             Exception: If the pipeline run fails.
         """
+        # First check whether the code is running in a notebook.
+        if Environment.in_notebook():
+            raise RuntimeError(
+                "The Kubernetes orchestrator cannot run pipelines in a notebook "
+                "environment. The reason is that it is non-trivial to create "
+                "a Docker image of a notebook. Please consider refactoring "
+                "your notebook cells into separate scripts in a Python module "
+                "and run the code outside of a notebook when using this "
+                "orchestrator."
+            )
         if deployment.schedule:
             logger.warning(
                 "Skypilot Orchestrator currently does not support the"
                 "use of schedules. The `schedule` will be ignored "
                 "and the pipeline will be run immediately."
             )
 
-        # Set up some variables for configuration
-        orchestrator_run_id = str(uuid4())
-        environment[
-            ENV_ZENML_SKYPILOT_ORCHESTRATOR_RUN_ID
-        ] = orchestrator_run_id
-
         settings = cast(
             SkypilotBaseOrchestratorSettings,
             self.get_settings(deployment),
         )
 
-        entrypoint = PipelineEntrypointConfiguration.get_entrypoint_command()
-        entrypoint_str = " ".join(entrypoint)
-        arguments = PipelineEntrypointConfiguration.get_entrypoint_arguments(
-            deployment_id=deployment.id
+        pipeline_name = deployment.pipeline_configuration.name
+        orchestrator_run_name = get_orchestrator_run_name(pipeline_name)
+
+        assert stack.container_registry
```

Comment:
Maybe preferable instead of assert that you actually do a check and `if not...` then raise an error etc